pkg/scalers/beanstalkd_scaler.go

```diff
@@ -0,0 +1,246 @@
+package scalers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	beanstalk "github.com/beanstalkd/go-beanstalk"
+	"github.com/go-logr/logr"
+	"github.com/mitchellh/mapstructure"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	"github.com/kedacore/keda/v2/pkg/util"
+)
+
+const (
+	beanstalkdJobsMetricName                   = "jobs"
+	beanstalkdValueConfigName                  = "value"
+	beanstalkdActivationValueTriggerConfigName = "activationValue"
+	beanstalkdMetricType                       = "External"
+	beanstalkdNetworkProtocol                  = "tcp"
+)
+
+type beanstalkdScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *beanstalkdMetadata
+	connection *beanstalk.Conn
+	tube       *beanstalk.Tube
+	logger     logr.Logger
+}
+
+type beanstalkdMetadata struct {
+	server          string
+	tube            string
+	value           float64
+	activationValue float64
+	includeDelayed  bool
+	timeout         time.Duration
+	triggerIndex    int
+}
+
+// TubeStats represents a set of tube statistics.
+type tubeStats struct {
+	TotalJobs    int64 `mapstructure:"total-jobs"`
+	JobsReady    int64 `mapstructure:"current-jobs-ready"`
+	JobsReserved int64 `mapstructure:"current-jobs-reserved"`
+	JobsUrgent   int64 `mapstructure:"current-jobs-urgent"`
+	JobsBuried   int64 `mapstructure:"current-jobs-buried"`
+	JobsDelayed  int64 `mapstructure:"current-jobs-delayed"`
+}
+
+func NewBeanstalkdScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	s := &beanstalkdScaler{}
+
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+	s.metricType = metricType
+
+	s.logger = InitializeLogger(config, "beanstalkd_scaler")
+
+	meta, err := parseBeanstalkdMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing beanstalkd metadata: %w", err)
+	}
+	s.metadata = meta
+
+	conn, err := beanstalk.DialTimeout(beanstalkdNetworkProtocol, s.metadata.server, s.metadata.timeout)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to beanstalkd: %w", err)
+	}
+
+	s.connection = conn
+
+	s.tube = beanstalk.NewTube(s.connection, meta.tube)
+
+	return s, nil
+}
+
+func parseBeanstalkdMetadata(config *scalersconfig.ScalerConfig) (*beanstalkdMetadata, error) {
```

Comment:
KEDA started implementing a new method of parsing metadata using struct field tags. It's recommended to parse metadata this way with new scalers. Here's an [example](https://github.com/kedacore/keda/blob/main/pkg/scalers/splunk_scaler.go#L27-L38) of use of the tags and [here](https://github.com/kedacore/keda/blob/main/pkg/scalers/splunk_scaler.go#L77) is how parsing is invoked. Not saying what you're doing is won't work, but I expect the maintainers will make this recommendation and they haven't reviewed your contribution yet.

Reply:
Thanks for the review 😄 I've updated the metadata parsing to use the updated method now.